### PR TITLE
fix(http): make URM client use userID in filter on find

### PR DESCRIPTION
**Pull request by affo**
Why it doesn't?

### CODE REVIEW

File: http/user_resource_mapping_service.go [hunk 1]
```diff
@@ -278,13 +278,15 @@ func (s *UserResourceMappingService) FindUserResourceMappings(ctx context.Contex
...
-	urs := make([]*influxdb.UserResourceMapping, len(results.Users))
-	for k, item := range results.Users {
-		urs[k] = &influxdb.UserResourceMapping{
-			ResourceID:   f.ResourceID,
-			ResourceType: f.ResourceType,
-			UserID:       item.User.ID,
-			UserType:     item.Role,
+	urs := make([]*influxdb.UserResourceMapping, 0, len(results.Users))
+	for _, item := range results.Users {
+		if !f.UserID.Valid() || f.UserID != item.User.ID {
```
Comment (jsteenb2) [inline http/user_resource_mapping_service.go]: where is the test for this?

Comment (affo) [reply http/user_resource_mapping_service.go]: For the client?

There's no test for the http client I think

Comment (affo) [reply http/user_resource_mapping_service.go]: I can create one as part of this PR 👍

Comment (jsteenb2) [reply http/user_resource_mapping_service.go]: That brings up another question, why isn't the server doing the filtering?

File: http/document_test.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ var (
 		},
 		Content: "content5",
 	}
-	doc6        = influxdb.Document{
+	doc6 = influxdb.Document{
```
Comment (affo) [inline http/document_test.go]: go fmt

File: http/user_resource_mapping_client_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		log:                        zaptest.NewLogger(t),
+		ResourceType:               rt,
+		UserType:                   ut,
+		UserResourceMappingService: mock.NewUserResourceMappingService(),
+		UserService:                userSvc,
+	}
+	h := OwnerMemberHandler{Router: NewRouter(backend.HTTPErrorHandler)}
+	h.HandlerFunc("POST", idPath, newPostMemberHandler(backend))
+	h.HandlerFunc("GET", idPath, newGetMembersHandler(backend))
+	h.HandlerFunc("GET", userIDPath, newGetMemberHandler(backend))
```
Comment (affo) [inline http/user_resource_mapping_client_test.go]: plan to make this a general util to be reused across handlers 👍

File: http/user_resource_mapping_service.go [hunk 4]
```diff
@@ -203,6 +204,105 @@ func decodeGetMembersRequest(ctx context.Context, r *http.Request) (*getMembersR
...
+			}, w)
+			return
+		}
+		m := mappings[0]
+		u, err := b.UserService.FindUserByID(ctx, m.UserID)
+		if err != nil {
+			b.HandleHTTPError(ctx, err, w)
+			return
+		}
+		b.log.Debug("Member/owner retrieved", zap.String("user", fmt.Sprint(u)))
```
Comment (jsteenb2) [inline http/user_resource_mapping_service.go]: what is the debug message for?

**Pull request closed without merging**